### PR TITLE
Feature/generic converter

### PR DESCRIPTION
**Pull request by bjarneo**
This PR will
- Add a generic converter which accepts other converters
- Add basic documentation for converters

### CODE REVIEW

File: src/converters/convert.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/**
+ * Given an input is a converter, it will return a converted result.
+ *
+ * @param {object} input - The input to be converted.
+ * @param {boolean} defaultValue - Default value to use if conversion fails.
+ * @param {string} name - The name of of what is converted.
+ *
+ * @returns {object} - The converted result.
+ */
```
Comment (andreasrs) [inline src/converters/convert.js:7]: This is the esdocs for the function that this function returns. Is this the correct way of documenting this?

File: src/converters/convert.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+export default function convert(...converters) {
+    if (!converters.length) {
+        throw new Error('You need to use at least one converter.');
+    }
+
+    return (input, defaultValue, name) => {
+        const result = converters.map((converter) => {
+            return converter(input, defaultValue, name);
```
Comment (andreasrs) [inline src/converters/convert.js]: This can be written as 
const result = converters.map((converter) => converter(input, defaultValue, name));

**Comment by andreasrs** [comment]
Some minor comments. Otherwise 👍  from me. Kudos for docs and tests.

File: src/converters/convert.js (hunk 1 above)
Comment (bjarneo) [inline src/converters/convert.js:7]: The most correct approach would be @returns {string|object[]|bool|number|RegExp} or undefined. Not sure if it should be a more generic return, or just write everything.

Comment (dlmr) [inline src/converters/convert.js:7]: I think Andreas is talking about `@params`, they seem to be copy pasted.

Comment (bjarneo) [inline src/converters/convert.js:7]: Right. They are. Forgot to update them. Will fix.

**Comment by bjarneo** [comment]
Concerns solved

File: src/converters/convert.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * @returns {object} - The converted result.
+ */
+export default function convert(...converters) {
+    if (!converters.length) {
+        throw new Error('You need to use at least one converter.');
+    }
+
+    return (input, defaultValue, name) => {
+        const result = converters.map((converter) => converter(input, defaultValue, name));
```
Comment (dlmr) [inline src/converters/convert.js]: Can we modify this so we only use a converter if the previous one failed, basically not precompute the result array.

**Pull request merged**